2025-03-09 22:15:20,756 INFO    Thread-21 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221520-i5wp5tk3\logs\debug.log
2025-03-09 22:15:20,761 INFO    Thread-21 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221520-i5wp5tk3\logs\debug-internal.log
2025-03-09 22:15:20,763 INFO    Thread-21 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:15:20,764 INFO    Thread-21 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'relu', 'batchSize': 132, 'hiddenLayer_Size': 32, 'learningRate': 0.001, 'num_epochs': 10, 'num_hiddenLayers': 5, 'optimizer': 'nesterov', 'weightDecay': 0, 'weightInit': 'random'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:15:20,765 INFO    Thread-21 (_run_job):1332 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:15:20,907 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/i5wp5tk3
2025-03-09 22:15:20,908 ERROR   Thread-21 (_run_job):1332 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:15:20,908 INFO    Thread-21 (_run_job):1332 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:15:20,908 INFO    Thread-21 (_run_job):1332 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:15:20,908 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:15:20,908 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_restore():2322] restore
2025-03-09 22:15:20,908 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:15:22,938 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:15:22,938 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:15:22,938 INFO    Thread-21 (_run_job):1332 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:15:22,947 ERROR   MainThread:1332 [pyagent.py:_run_jobs_from_queue():234] Run i5wp5tk3 errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2024573097.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2024573097.py", line 133, in backpropagation
    deltas.append(deltas[-1].dot(self.weights[i].T) * self.sigmoid_derivative(self.layers[i]))
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ValueError: operands could not be broadcast together with shapes (132,32) (32,132) 

2025-03-09 22:15:22,948 ERROR   MainThread:1332 [pyagent.py:_run_jobs_from_queue():245] Detected 3 failed runs in the first 60 seconds, killing sweep.
2025-03-09 22:17:08,725 INFO    MainThread:1332 [pyagent.py:run():322] Starting sweep agent: entity=None, project=None, count=200
2025-03-09 22:17:09,876 INFO    MsgRouterThr:1332 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_setup.py:_flush():67] Configure stats pid to 1332
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from C:\Users\prajv\.config\wandb\settings
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\settings
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221711-lj96nef9\logs\debug.log
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221711-lj96nef9\logs\debug-internal.log
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:monkeypatch_ipython():599] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x0000026DC5360A40>
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 10, 'num_hiddenLayers': 4, 'optimizer': 'nesterov', 'weightDecay': 0.5, 'weightInit': 'xavier'}
config: {'_wandb': {}}
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():784] starting backend
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():788] sending inform_init request
2025-03-09 22:17:11,225 INFO    Thread-28 (_run_job):1332 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=spawn, using: spawn
2025-03-09 22:17:11,235 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():798] backend started and connected
2025-03-09 22:17:11,236 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_config_callback():1261] config_cb None None {'activationFunc': 'tanh', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 10, 'num_hiddenLayers': 4, 'optimizer': 'nesterov', 'weightDecay': 0.5, 'weightInit': 'xavier'}
2025-03-09 22:17:11,237 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_label_probe_notebook():1204] probe notebook
2025-03-09 22:17:11,261 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():891] updated telemetry
2025-03-09 22:17:11,324 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-09 22:17:13,160 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():990] starting run threads in backend
2025-03-09 22:17:13,446 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_console_start():2375] atexit reg
2025-03-09 22:17:13,447 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-09 22:17:13,448 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-09 22:17:13,448 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-09 22:17:13,450 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-09 22:17:13,452 INFO    Thread-28 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221713-lj96nef9\logs\debug.log
2025-03-09 22:17:13,458 INFO    Thread-28 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221713-lj96nef9\logs\debug-internal.log
2025-03-09 22:17:13,458 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:17:13,458 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'tanh', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.0001, 'num_epochs': 10, 'num_hiddenLayers': 4, 'optimizer': 'nesterov', 'weightDecay': 0.5, 'weightInit': 'xavier'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:17:13,458 INFO    Thread-28 (_run_job):1332 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:17:13,634 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/lj96nef9
2025-03-09 22:17:13,636 ERROR   Thread-28 (_run_job):1332 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:17:13,636 INFO    Thread-28 (_run_job):1332 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:17:13,637 INFO    Thread-28 (_run_job):1332 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:17:13,637 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:17:13,637 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_restore():2322] restore
2025-03-09 22:17:13,637 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:17:15,724 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:17:15,725 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:17:15,725 INFO    Thread-28 (_run_job):1332 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:17:15,731 ERROR   MainThread:1332 [pyagent.py:_run_jobs_from_queue():234] Run lj96nef9 errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2379888119.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2379888119.py", line 133, in backpropagation
    deltas.append(deltas[-1].dot(self.weights[i].T).T * self.sigmoid_derivative(self.layers[i]))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (64,164) and (64,64) not aligned: 164 (dim 1) != 64 (dim 0)

2025-03-09 22:17:20,752 INFO    MsgRouterThr:1332 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-09 22:17:21,762 INFO    Thread-33 (_run_job):1332 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-09 22:17:21,762 INFO    Thread-33 (_run_job):1332 [wandb_setup.py:_flush():67] Configure stats pid to 1332
2025-03-09 22:17:21,762 INFO    Thread-33 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from C:\Users\prajv\.config\wandb\settings
2025-03-09 22:17:21,762 INFO    Thread-33 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\settings
2025-03-09 22:17:21,762 INFO    Thread-33 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221721-m2rgq3rg\logs\debug.log
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221721-m2rgq3rg\logs\debug-internal.log
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:monkeypatch_ipython():599] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x0000026DC4B51EB0>
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'relu', 'batchSize': 132, 'hiddenLayer_Size': 32, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 4, 'optimizer': 'sgd', 'weightDecay': 0.5, 'weightInit': 'xavier'}
config: {'_wandb': {}}
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():784] starting backend
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():788] sending inform_init request
2025-03-09 22:17:21,774 INFO    Thread-33 (_run_job):1332 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=spawn, using: spawn
2025-03-09 22:17:21,780 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():798] backend started and connected
2025-03-09 22:17:21,781 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_config_callback():1261] config_cb None None {'activationFunc': 'relu', 'batchSize': 132, 'hiddenLayer_Size': 32, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 4, 'optimizer': 'sgd', 'weightDecay': 0.5, 'weightInit': 'xavier'}
2025-03-09 22:17:21,783 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_label_probe_notebook():1204] probe notebook
2025-03-09 22:17:21,802 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():891] updated telemetry
2025-03-09 22:17:21,857 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-09 22:17:23,607 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():990] starting run threads in backend
2025-03-09 22:17:23,932 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_console_start():2375] atexit reg
2025-03-09 22:17:23,932 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-09 22:17:23,932 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-09 22:17:23,933 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-09 22:17:23,933 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-09 22:17:23,940 INFO    Thread-33 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221723-m2rgq3rg\logs\debug.log
2025-03-09 22:17:23,940 INFO    Thread-33 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221723-m2rgq3rg\logs\debug-internal.log
2025-03-09 22:17:23,942 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:17:23,942 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'relu', 'batchSize': 132, 'hiddenLayer_Size': 32, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 4, 'optimizer': 'sgd', 'weightDecay': 0.5, 'weightInit': 'xavier'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:17:23,942 INFO    Thread-33 (_run_job):1332 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:17:24,075 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/m2rgq3rg
2025-03-09 22:17:24,076 ERROR   Thread-33 (_run_job):1332 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:17:24,076 INFO    Thread-33 (_run_job):1332 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:17:24,076 INFO    Thread-33 (_run_job):1332 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:17:24,076 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:17:24,076 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_restore():2322] restore
2025-03-09 22:17:24,076 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:17:26,016 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:17:26,016 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:17:26,016 INFO    Thread-33 (_run_job):1332 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:17:26,032 ERROR   MainThread:1332 [pyagent.py:_run_jobs_from_queue():234] Run m2rgq3rg errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2379888119.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2379888119.py", line 133, in backpropagation
    deltas.append(deltas[-1].dot(self.weights[i].T).T * self.sigmoid_derivative(self.layers[i]))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (32,132) and (32,32) not aligned: 132 (dim 1) != 32 (dim 0)

2025-03-09 22:17:36,091 INFO    MsgRouterThr:1332 [mailbox.py:close():129] Closing mailbox, abandoning 1 handles.
2025-03-09 22:17:37,142 INFO    Thread-38 (_run_job):1332 [wandb_setup.py:_flush():67] Current SDK version is 0.19.8
2025-03-09 22:17:37,142 INFO    Thread-38 (_run_job):1332 [wandb_setup.py:_flush():67] Configure stats pid to 1332
2025-03-09 22:17:37,142 INFO    Thread-38 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from C:\Users\prajv\.config\wandb\settings
2025-03-09 22:17:37,142 INFO    Thread-38 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\settings
2025-03-09 22:17:37,142 INFO    Thread-38 (_run_job):1332 [wandb_setup.py:_flush():67] Loading settings from environment variables
2025-03-09 22:17:37,142 INFO    Thread-38 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221737-9dvj5ygb\logs\debug.log
2025-03-09 22:17:37,148 INFO    Thread-38 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221737-9dvj5ygb\logs\debug-internal.log
2025-03-09 22:17:37,148 INFO    Thread-38 (_run_job):1332 [wandb_init.py:monkeypatch_ipython():599] configuring jupyter hooks <wandb.sdk.wandb_init._WandbInit object at 0x0000026DC8829EB0>
2025-03-09 22:17:37,148 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:17:37,148 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'relu', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 3, 'optimizer': 'momentum', 'weightDecay': 0.0005, 'weightInit': 'xavier'}
config: {'_wandb': {}}
2025-03-09 22:17:37,148 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():784] starting backend
2025-03-09 22:17:37,150 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():788] sending inform_init request
2025-03-09 22:17:37,154 INFO    Thread-38 (_run_job):1332 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=spawn, using: spawn
2025-03-09 22:17:37,154 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():798] backend started and connected
2025-03-09 22:17:37,156 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_config_callback():1261] config_cb None None {'activationFunc': 'relu', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 3, 'optimizer': 'momentum', 'weightDecay': 0.0005, 'weightInit': 'xavier'}
2025-03-09 22:17:37,157 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_label_probe_notebook():1204] probe notebook
2025-03-09 22:17:37,176 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():891] updated telemetry
2025-03-09 22:17:37,241 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():915] communicating run to backend with 90.0 second timeout
2025-03-09 22:17:39,169 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():990] starting run threads in backend
2025-03-09 22:17:39,439 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_console_start():2375] atexit reg
2025-03-09 22:17:39,440 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_redirect():2227] redirect: wrap_raw
2025-03-09 22:17:39,440 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_redirect():2292] Wrapping output streams.
2025-03-09 22:17:39,441 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_redirect():2315] Redirects installed.
2025-03-09 22:17:39,443 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():1032] run started, returning control to user process
2025-03-09 22:17:39,443 INFO    Thread-38 (_run_job):1332 [wandb_init.py:setup_run_log_directory():647] Logging user logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221739-9dvj5ygb\logs\debug.log
2025-03-09 22:17:39,443 INFO    Thread-38 (_run_job):1332 [wandb_init.py:setup_run_log_directory():648] Logging internal logs to c:\Users\prajv\Downloads\ass1\DA6401_ASSIGNMENT1\wandb\run-20250309_221739-9dvj5ygb\logs\debug-internal.log
2025-03-09 22:17:39,443 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():761] calling init triggers
2025-03-09 22:17:39,443 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():766] wandb.init called with sweep_config: {'activationFunc': 'relu', 'batchSize': 164, 'hiddenLayer_Size': 64, 'learningRate': 0.001, 'num_epochs': 20, 'num_hiddenLayers': 3, 'optimizer': 'momentum', 'weightDecay': 0.0005, 'weightInit': 'xavier'}
config: {'num_epochs': 10, 'num_hiddenLayers': 3, 'hiddenLayer_Size': 40, 'weightDecay': 0, 'learningRate': 0.001, 'optimizer': 'Nesterov_Accelerated_GD', 'batchSize': 32, 'weightInit': 'xavier', 'activationFunc': 'sigmoid', '_wandb': {}}
2025-03-09 22:17:39,443 INFO    Thread-38 (_run_job):1332 [wandb_init.py:init():776] wandb.init() called while a run is active
2025-03-09 22:17:39,577 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_finish():2112] finishing run da6401-assignments/assignment1/9dvj5ygb
2025-03-09 22:17:39,579 ERROR   Thread-38 (_run_job):1332 [jupyter.py:save_history():450] Run pip install nbformat to save notebook history
2025-03-09 22:17:39,579 INFO    Thread-38 (_run_job):1332 [jupyter.py:save_ipynb():386] not saving jupyter notebook
2025-03-09 22:17:39,579 INFO    Thread-38 (_run_job):1332 [wandb_init.py:_jupyter_teardown():581] cleaning up jupyter logic
2025-03-09 22:17:39,580 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_atexit_cleanup():2340] got exitcode: 1
2025-03-09 22:17:39,580 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_restore():2322] restore
2025-03-09 22:17:39,581 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_restore():2328] restore done
2025-03-09 22:17:41,522 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3956] rendering history
2025-03-09 22:17:41,522 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_footer_history_summary_info():3988] rendering summary
2025-03-09 22:17:41,522 INFO    Thread-38 (_run_job):1332 [wandb_run.py:_footer_sync_info():3917] logging synced files
2025-03-09 22:17:41,527 ERROR   MainThread:1332 [pyagent.py:_run_jobs_from_queue():234] Run 9dvj5ygb errored:
Traceback (most recent call last):
  File "c:\Python312\Lib\site-packages\wandb\agents\pyagent.py", line 306, in _run_job
    self._function()
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\1058529509.py", line 36, in sweep_hyperParameters
    model.train(x_train, y_train, x_val, y_val, num_epochs, learningRate, optimizer, batchSize)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2379888119.py", line 61, in train
    self.backpropagation(x_batch, y_batch, learning_rate, optimizer)
  File "C:\Users\prajv\AppData\Local\Temp\ipykernel_1332\2379888119.py", line 133, in backpropagation
    deltas.append(deltas[-1].dot(self.weights[i].T).T * self.sigmoid_derivative(self.layers[i]))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: shapes (64,164) and (64,64) not aligned: 164 (dim 1) != 64 (dim 0)

2025-03-09 22:17:41,527 ERROR   MainThread:1332 [pyagent.py:_run_jobs_from_queue():245] Detected 3 failed runs in the first 60 seconds, killing sweep.
